webClient/src/app/shared/editor-control/editor-control.service.ts

```diff
@@ -583,6 +584,126 @@ export class EditorControlService implements ZLUX.IEditor, ZLUX.IEditorMultiBuff
+    //TODO: validation of record length that is aware of how DBCS will effect actual length
+    //FB must have exactly lrecl, VB must have no more than lrecl. content cant be undefined.
+    const MAX_CONTENT_LENGTH = 5242880;
+    if (!content) {
+      return 'No Content';
+    }
+    let maxRecordLen = datasetAttrs.dsorg.maxRecordLen;
+    for (let i = 0; i < content.length; i++) {
+      let record = content[i];
+      if (record.length > maxRecordLen) {
+        return `Line ${i} exceeds record length limit of ${maxRecordLen}`;
+      }
+    }
+    if (JSON.stringify({records:content}).length > MAX_CONTENT_LENGTH) {
+      return 'Content over max size currently supported ('+ (MAX_CONTENT_LENGTH/(1024*1024)) +'MB)';
+    }
+    return true;
+  }
+  
+  saveDatasetHandler(context?: ProjectContext, destinationOverride?: any): Observable<void> {
+    const _openDataset = this.openFileList.getValue();
+    const editor = this._editor.getValue();
+    let _activeDataset: ProjectContext;
+    const forceWrite = false;
+    let contents;
+    if (editor) {
+      contents = editor.getValue();
+    }
+    if (context != null) {
+      _activeDataset = context;
+    } else {
+      _activeDataset = _openDataset.filter(dataset => dataset.active === true)[0];
+    }
+    const model = _activeDataset.model;
+    const isDataset = _activeDataset.model.isDataset;
+    const fullName = isDataset ? _activeDataset.model.fileName : _activeDataset.model.name
+    if (!isDataset) {
+      this.snackBar.open(`${fullName} is not a dataset, invalid save route`, "Close", { duration: MessageDuration.Medium, panelClass: 'center' });
+      return;
+    }
+    if (!editor || contents === undefined) {
+      this.snackBar.open(`Cannot get editor or no contents to save`, "Close", {duration: MessageDuration.Medium, panelClass: 'center'});
+      return;
+    }
+
+    if (!destinationOverride && fullName) {
+
+      contents = editor.getValue().split('\n');
+      const requestUrl = ZoweZLUX.uriBroker.datasetContentsUri(fullName);
+      this.log.debug(`Should save contents to dataset. dataset=${fullName}, route=${requestUrl}`);
+      let result = this.isContentValidForDataset(contents, model.datasetAttrs);
+      if (result === true) {
+        this.saveDataset(context, _activeDataset, forceWrite);  
+      } else {
+        this.snackBar.open(`Content invalid for saving to dataset. Reason=${result}`, 'Close', { duration:MessageDuration.Long, panelClass: 'center'});
+      }
+    } else {
+      //dataset is new or needs some alteration we can't do yet
+      this.snackBar.open(`${_activeDataset.name} could not be saved, feature not yet implemented.`, 
+                         'Close', { duration: MessageDuration.Long,   panelClass: 'center' });    
+    }
+  }
+
+  saveDataset(context: ProjectContext, activeDataset: ProjectContext, forceWrite: Boolean): void {
+    const editor = this._editor.getValue();
```

Comment:
So save works for me 👍 both Ctrl+S and menu key, but I see some other strange network calls going on with /unixfile/metadata which doesn't make sense to me because these are datasets, it shouldn't make any calls to unixfile. Open the network view and see if you get this too

![image](https://user-images.githubusercontent.com/20528015/134260712-8044ad57-bcd4-4b9f-9456-aa4d7d7c9b28.png)

Reply:
It looks like it's calling `getFileMetadata `in zlux-editor\webClient\src\app\shared\editor-control\editor-control.service.ts for some reason